packages/cli/src/lib/bundler/config.ts

```diff
@@ -206,6 +211,16 @@ export async function createConfig(
         : {}),
     },
     plugins,
+    ...(withCache
+      ? {
+          cache: {
+            type: 'filesystem',
+            buildDependencies: {
+              config: [__filename],
```

Comment:
This one is probably gonna be tricky, since it's likely not the correct file in the published package. You can see this in action if you log this, build the CLI, and then run it after replacing the bin entrypoint:

```diff
diff --git /packages/cli/bin/backstage-cli
-  require('../src');
+  require('../dist/index.cjs');
```

I think what we might need to do here is to list out the directory contents and use all of the files as the hash input.

Reply:
Ahh, nice catch 👍 Reading a bit more [here](https://github.com/webpack/changelog-v5/blob/master/guides/persistent-caching.md) it seems that we can provide a directory (ending with a slash).

Reply:
Oh what, reading through that I think your original implementation might actually be the best one. Seems like there's quite a lot of magic going for webpack to figure out the dependencies. As far as I can tell specifying `__filename` will lead to any other required modules also being part of the cache key, which is exactly what we want

Reply:
Ended up reverting so that we can :shipit:. If it doesn't work we'll fix 😁